R/repo.R

```diff
@@ -63,15 +63,17 @@ package_repo <- function(desc, meta) {
     return(meta[["repo"]])
   }
 
-  # Otherwise try and guess from BugReports + URLs
+  # Otherwise try and guess from URLs + BugReports
   urls <- c(
-    desc$get_field("BugReports", default = character()),
-    desc$get_urls()
+    desc$get_urls(),
```

Comment:
I think this change is too risky (since it is likely to change packages in the wild), and doesn't seem that important. Can you please revert?

Reply:
I've reverted this change.

---

But for the record: The current behaviour of preferring the `BugReports` field over `URL` leads to potential ambiguities with GitLab projects: GitLab generally allows to create a (sub)project named `issues`. So if someone decides to maintain a single repository for tracking issues in, say, a collection of interrelated R packages, he might have the following `DESCRIPTION` file:

```deb
URL: https://gitlab.com/someone/r/pkgs/a-pkg
BugReports: https://gitlab.com/someone/r/pkgs/issues
```

Now current pkgdown runs into 2 problems with the above:

1. Preferring the `BugReports` URL results in wrong code repo links.

2. It cannot know if the `BugReports` URL links to an actual subproject named `issues` (with repo path `r/pkgs/issues`) or if the URL directly links to the bug tracker of a subproject named `pkgs` (with repo path `r/pkgs`).

    My PR changes the code to always assume the latter if a `BugReports` URL ends in `issues` or `issues/` which should be fine for the majority of users.

These might be "luxury problems", easy to manually correct afterwards. But I still think generally preferring the `BugReports` field over the `URL` field to determine the code repo URL is not a particularly good design choice.

Reply:
In that case, I think it's better to rely on manual specification: https://pkgdown.r-lib.org/dev/reference/build_site.html#yaml-config-repo